### PR TITLE
Make `insert_non_basic_units` a method

### PR DESCRIPTION
**Pull request by younies**

### CODE REVIEW

File: components/experimental/src/units/converter_factory.rs [hunk 1]
```diff
@@ -38,6 +38,21 @@ impl From<Sign> for num_bigint::Sign {
     }
 }
 
+/// A struct that contains the sum and difference of base unit powers.
```
Comment (robertbastian) [inline components/experimental/src/units/converter_factory.rs:41]: nit: keep this in the method

Comment (younies) [reply components/experimental/src/units/converter_factory.rs:41]: unfortunatley, it is used as output that is used in other places

Notice: `map: &mut LiteMap<u16, PowersInfo>,`

Comment (robertbastian) [reply components/experimental/src/units/converter_factory.rs:41]: I don't think this PR is worth it then. `insert_non_basic_units` is an implementation detail of `is_reciprocal`, and `PowersInfo` is an implementation detail of that. Having them both at the top level is not well encapsulated.

**Review by robertbastian — COMMENTED** [review]
.

**Pull request closed without merging**